File: node/resources.js
const axios = require("axios");
const { response } = require("express");
const randomNumber = require("random-number-csprng");

const letters = "abcdefghijklmnopqrstuvwxyz";
const passphraseSize = 5;

class PassPhrase {
  constructor(passphrase) {
    this.passphrase = passphrase;
  }
}

const url = (word) => `http://api.datamuse.com/words?sp=${word}*`;

const randomItem = async (arr, max = null) => {
  return arr[await randomNumber(0, (max || arr.length) - 1)];
};

const wordRequest = async () => {
  let response = await axios.get(url(await randomItem(letters)));
  return await randomItem(response.data);
};

const getPassPhrase = async () => {
  let words = [];
  const insertIntoWords = async () => {
    let {word} = await wordRequest()
    words.push(word);
  };
  const proms = [];
  for (let i = 0; i < passphraseSize; i++) {
    proms.push(insertIntoWords());
  }
  await Promise.all(proms);

  for (let i = words.length - 1; i > 0; i--) {
    const j = await randomNumber(0, i);
    const temp = words[i];
    words[i] = words[j];
    words[j] = temp;
  }
  return new PassPhrase(words.join(" "));
};

module.exports = {
  getPassPhrase: getPassPhrase,
};

File: python_fastapi/src/resources.py
import asyncio
from typing import List
from pydantic import BaseModel
from aiohttp import ClientSession
from secrets import randbelow
from random import shuffle


LETTERS = "abcdefghijklmnopqrstuvwxyz"
PASSPHASE_SIZE = 5


class PassPhrase(BaseModel):
    passphrase: str


URL = "http://api.datamuse.com/words?sp={letter}*"


def random_item(list_, max_):
    return list_[randbelow(max_)]


async def word_request():
    async with ClientSession() as session:
        async with session.get(URL.format(letter=random_item(LETTERS, 26))) as response:
            data = await response.json()
            return random_item(data, len(data))

async def get_passphrase():
    words: List[str] = []
    for cor in asyncio.as_completed([word_request() for _ in range(PASSPHASE_SIZE)]):
        words.append((await cor).get("word"))
    shuffle(words)
    return " ".join(words)
    
